extensions/meme-generator/src/components/MemeForm.tsx

```diff
@@ -31,18 +32,19 @@ export default function MemeForm({ id, title, boxCount }: Meme) {
 
       generateMeme({ id, boxes })
         .then(async (results) => {
-          copyFileToClipboard(results.url, `${title}.jpg`).then((file) => {
-            toast.hide();
-            showHUD(`Meme "${file}" copied to clipboard`);
-            closeMainWindow();
-            pop();
-          });
+          if (preview) {
+            push(<MemePreview title={title} url={results.url} />);
+          } else {
+            await copyFileToClipboard(results.url, `${title}.jpg`);
+            showHUD(`Meme "${title}" copied to clipboard`);
```

Comment:
You can use `Toasts` (instead of HUD) to produce a colorful prompt.
```suggestion
            await closeMainWindow();
            await showToast(Toast.Style.Success, `Meme "${title}" copied to clipboard`);
```